File: task 23_02_2025/financial_calculator.py
loan_amount = float(input("Enter the loan amount: "))
loan_period = int(input("Enter the loan period: "))

months = loan_period*12
rate = 5.0
print(f"{"Monthly Rate": <15}{"Monthly Payment": <18}{"Total Payment": <15}")
while (rate <= 10.0):

	monthly_rate = rate / (100 * 12)

	monthly_payment = loan_amount * monthly_rate * ((1 + monthly_rate) ** months) * (1 / (((1 + monthly_rate) ** months) - 1))
	total_payment = monthly_payment * months

	print(f"{rate: <15}{round(monthly_payment, 2): <18}{round(total_payment, 2): <15}")
	rate += 0.25

File: task 23_02_2025/FinancialCalculator.java
import java.util.Scanner;
public class FinancialCalculator{

	public static void main (String [] args){

		Scanner input = new Scanner (System.in);

		System.out.println("Enter loan amount: ");
		double loanAmount = input.nextDouble();

		System.out.println("Enter number of years: ");
		int years = input.nextInt();

		int months = years * 12;

		System.out.printf("%-15s%-18s%-15s%n", "Interest Rate", "Monthly Payment", "Total Payment");

		for (double rate = 5.0; rate <= 10; rate += 0.25){

			double monthlyRate = rate/(100 * 12);

			double monthlyPayment = loanAmount * monthlyRate * Math.pow((1 + monthlyRate), months) * (1 / (Math.pow((1 + monthlyRate), months) - 1));
			double totalPayment = monthlyPayment * months;

			System.out.printf("%-15.2f%-18.2f%-15.2f%n", rate, monthlyPayment, totalPayment);

		}

		
	}

	

}